### PR TITLE
Add table function file

### PR DESCRIPTION
**Pull request by decaseal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

**Description**
file(path, format, structure) - creates a temporary StorageFile.
The file must be in the data directory on clickhouse server.
The relative path begins with the data directory on clickhouse server.

**Sample**
`SELECT * FROM file('test.tsv', 'TabSeparated', 'id UInt64, name String')`

### CODE REVIEW

**Comment by decaseal** [comment]
**Changes in StorageFile**
Сreating a table from a file in the current database directory now allowed.

File: dbms/src/Storages/StorageFile.cpp [hunk 1]
```diff
@@ -41,10 +44,21 @@ static std::string getTablePath(const std::string & db_dir_path, const std::stri
     return db_dir_path + escapeForFileName(table_name) + "/data." + escapeForFileName(format_name);
 }
 
-static void checkCreationIsAllowed(Context & context_global)
+static void checkCreationIsAllowed(Context & context_global, const std::string & db_dir_path, const std::string & table_path, const int & table_fd)
```
Comment (alexey-milovidov) [inline dbms/src/Storages/StorageFile.cpp]: Why `const int & table_fd`?

File: dbms/src/Storages/StorageFile.cpp [hunk 2]
```diff
@@ -65,8 +80,12 @@ StorageFile::StorageFile(
 
         if (!table_path_.empty()) /// Is user's file
         {
-            checkCreationIsAllowed(context_global);
-            path = Poco::Path(table_path_).absolute().toString();
+            Poco::Path poco_path = Poco::Path(table_path_);
+            if (poco_path.isRelative())
```
Comment (alexey-milovidov) [inline dbms/src/Storages/StorageFile.cpp]: What if path contain `..`?

**Comment by alexey-milovidov** [comment]
Security issue: the user can exploit storage `File` and table function `file` to read arbitary data from databases without access granted.

**Comment by alexey-milovidov** [comment]
You can add integration test for this feature. Look at `tests/integration`.

**Comment by alexey-milovidov** [comment]
> Security issue

Possible solution: allow to read files only from inside `path/files` directory.
(`files` subdirectory will be reserved for user provided files for tables)

**Comment by alexey-milovidov** [comment]
Now you should place your files into path/`user_files/` directory.

**Pull request merged**